lib/presentation/history_screen.dart

```diff
@@ -72,7 +75,24 @@ class _HistoryScreenState extends State<HistoryScreen> {
             );
           }
 
-          final transactions = state.transactions;
+          final transactions = selectedItems.isEmpty
+              ? state.transactions
+              : state.transactions.where((transaction) {
+                  final selectedCategories =
+                      selectedItems.whereType<CategoryEntity>().toList();
+                  final selectedWallets =
+                      selectedItems.whereType<WalletEntity>().toList();
+
+                  final bool categoryMatch = selectedCategories.isNotEmpty ||
+                      transaction.categories.any((cat) => selectedCategories
+                          .any((selected) => selected.id == cat.id));
+
+                  final bool walletMatch = selectedWallets.isNotEmpty ||
+                      selectedWallets
+                          .any((wallet) => wallet.id == transaction.wallet.id);
```

Comment:
```suggestion
                  final bool categoryMatch = selectedCategories.isNotEmpty &&
                      transaction.categories.any((cat) => selectedCategories
                          .any((selected) => selected.id == cat.id));

                  final bool walletMatch = selectedWallets.isNotEmpty &&
                      selectedWallets
                          .any((wallet) => wallet.id == transaction.wallet.id);
```

Shouldn't these conditions `be` && instead of `||`?